src/Development/Shake/Internal/Core/Rules.hs

```diff
@@ -160,14 +166,15 @@ data SRules = SRules
     ,builtinRules :: !(Map.HashMap TypeRep{-k-} BuiltinRule)
     ,userRules :: !(TMap.Map UserRuleVersioned)
     ,targets :: !(ListBuilder Target)
+    ,helpSuffix :: !String
```

Comment:
This should be a `ListBuilder String` so we never get into quadratic append problems.